File: 253_meeting_rooms_ii.java
/**
 * Definition for an interval.
 * public class Interval {
 *     int start;
 *     int end;
 *     Interval() { start = 0; end = 0; }
 *     Interval(int s, int e) { start = s; end = e; }
 * }
 */
class Solution {
    public int minMeetingRooms(Interval[] intervals) {
        int length = intervals.length;
        int[] starts = new int[length];
        int[] ends = new int[length];
        int counter = 0, start = 0, end = 0;
        
        for (int i = 0; i < length; i++) {
            Interval interval = intervals[i];
            starts[i] = interval.start;
            ends[i] = interval.end;
        }
        
        Arrays.sort(starts);
        Arrays.sort(ends);
                
        for (; start < length; start++) {
            if (starts[start] < ends[end])
                counter++;
            else 
                end++;
        }
        
        return counter;
    }
}

File: 253_meeting_rooms_ii.py
# Definition for an interval.
# class Interval:
#     def __init__(self, s=0, e=0):
#         self.start = s
#         self.end = e

class Solution:
    def minMeetingRooms(self, intervals):
        """
        :type intervals: List[Interval]
        :rtype: int
        """
        starts, ends = [], []
        for it in intervals:
            starts.append(it.start)
            ends.append(it.end)
        
        starts.sort(); ends.sort()
        
        end_idx, counter = 0, 0
        for s in starts:
            if s >= ends[end_idx]:
                end_idx += 1
                continue
            else:
                counter += 1
        
        return counter
